src/pt-functions.lisp

```diff
@@ -194,10 +194,32 @@ font, size and spacing. Allocates a new RL-VECTOR2 unless you pass one."
 
 ;; Model loading/unloading functions
 
-(defun-pt load-model-from-mesh claylib/ll:load-model-from-mesh
+;; TODO generalize for all functions that do model creation
+(defun load-model-from-mesh (mesh &key (model (make-instance 'model)))
   "Load a model from a passed-in mesh. Allocates a new RL-MODEL unless you pass one."
-  (model rl-model nil (make-instance 'rl-model))
-  (mesh rl-mesh))
+  (check-type mesh rl-mesh)
+  (check-type model rl-model)
+  (claylib/ll:load-model-from-mesh (c-struct model) (c-struct mesh))
+  (let ((c-meshes (autowrap:c-aref (model.meshes (c-struct model)) 0 'claylib/ll:mesh))
+        (c-bones (autowrap:c-aref (model.bones (c-struct model)) 0 'claylib/ll:bone-info))
+        (c-materials (autowrap:c-aref (model.materials (c-struct model)) 0 'claylib/ll:material))
+        (c-poses (autowrap:c-aref (model.bind-pose (c-struct model)) 0 'claylib/ll:transform)))
+    (set-slot :transform model (transform model))
+    (setf (meshes model)
+          (make-instance 'rl-meshes :cl-array (make-rl-*-array c-meshes (mesh-count model)))
+
+          (materials model)
+          (make-instance 'rl-materials
+                         :cl-array (make-rl-*-array c-materials (material-count model)))
+
+          ;; (bones model)
+          ;; (make-instance 'rl-bones :cl-array (make-rl-*-array c-bones (bone-count model)))
+
+          ;; (bind-pose model)
+          ;; (make-instance 'rl-transforms :cl-array (make-rl-*-array c-poses (bone-count model)))
```

Comment:
FWIW, not getting bones or a bind pose here is expected. I don't think there's any scenario in which an orphaned mesh would have animation data directly associated with it.

Reply:
Right, this is what I thought but I wasn't sure. Then it should be fine to have a model with these slots unbound sometimes, in which case this should work exactly as intended.

Reply:
Fixed by https://github.com/defun-games/claylib/pull/58/commits/803c8dbb6f1e4291aedd9650fdc2865ee1a36e83